lib/react_on_rails/helper.rb

```
@@ -124,29 +124,72 @@ def react_component(component_name, options = {})
+      run_stream_inside_fiber do
+        internal_stream_react_component(component_name, options)
       end
+    end
 
-      rendering_fiber = Fiber.new do
-        stream = internal_stream_react_component(component_name, options)
-        stream.each_chunk do |chunk|
-          Fiber.yield chunk
-        end
+    # Renders the React Server Component (RSC) payload for a given component. This helper generates
+    # a special format designed by React for serializing server components and transmitting them 
```

RuboCop (1.69.1)

[convention] Trailing whitespace detected.

(Layout/TrailingWhitespace)